inference-engine/src/inference_engine/src/ie_core.cpp

```diff
@@ -864,63 +815,125 @@ class Core::Impl : public ICore, public std::enable_shared_from_this<ICore> {
      * @brief Provides a list of extensions
      * @return A list of registered extensions
      */
-    const std::vector<IExtensionPtr>& GetExtensions() const {
+    const std::vector<InferenceEngine::IExtensionPtr>& GetExtensions() const {
         return extensions;
     }
+
+    std::map<std::string, InferenceEngine::Version> GetVersions(const std::string& deviceName) const {
+        std::map<std::string, InferenceEngine::Version> versions;
+        std::vector<std::string> deviceNames;
+
+        {
+            // for compatibility with samples / demo
+            if (deviceName.find("HETERO") == 0) {
+                auto pos = deviceName.find_first_of(":");
+                if (pos != std::string::npos) {
+                    deviceNames = InferenceEngine::DeviceIDParser::getHeteroDevices(deviceName.substr(pos + 1));
+                }
+                deviceNames.push_back("HETERO");
+            } else if (deviceName.find("MULTI") == 0) {
+                auto pos = deviceName.find_first_of(":");
+                if (pos != std::string::npos) {
+                    deviceNames = InferenceEngine::DeviceIDParser::getMultiDevices(deviceName.substr(pos + 1));
+                }
+                deviceNames.push_back("MULTI");
+            } else if (deviceName.find("AUTO") == 0) {
+                auto pos = deviceName.find_first_of(":");
+                if (pos != std::string::npos) {
+                    deviceNames = InferenceEngine::DeviceIDParser::getHeteroDevices(deviceName.substr(pos + 1));
+                }
+                deviceNames.emplace_back("AUTO");
+            } else {
+                deviceNames.push_back(deviceName);
+            }
+        }
+
+        for (auto&& deviceName_ : deviceNames) {
+            InferenceEngine::DeviceIDParser parser(deviceName_);
+            std::string deviceNameLocal = parser.getDeviceName();
+
+            InferenceEngine::InferencePlugin cppPlugin = GetCPPPluginByName(deviceNameLocal);
+            const InferenceEngine::Version version = cppPlugin.GetVersion();
+            versions[deviceNameLocal] = version;
+        }
+
+        return versions;
+    }
 };
 
-Core::Core(const std::string& xmlConfigFile) {
-    _impl = std::make_shared<Impl>();
+}  // namespace core_detail
+using namespace core_detail;
```

Comment:
do we need this `using namespace`? maybe to use `core_detail` namespace directly for explicitness?